File: sources/WCSF2010/Trunk/Source/WebClientFactoryGP/WebClientFactoryPackage/bin/Release/Templates/Projects/Module.Tests.CS/ModuleInitializerFixture.cs
//===============================================================================
// Microsoft patterns & practices
// Web Client Software Factory 2010
//===============================================================================
// Copyright (c) Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
//===============================================================================
using System;
using System.Text;
using System.Collections.Generic;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Microsoft.Practices.CompositeWeb;
using Microsoft.Practices.CompositeWeb.Services;
using Microsoft.Practices.CompositeWeb.Interfaces;

namespace $ModuleTestProjectNamespace$
{
    /// <summary>
    /// Summary description for $ModuleName$ModuleInitializerFixture
    /// </summary>
    [TestClass]
    public class $ModuleName$ModuleInitializerFixture
    {
        public $ModuleName$ModuleInitializerFixture()
        {
        }        

        [TestMethod]
        public void $ModuleName$GetsRegisteredOnSiteMap()
        {
            TestableModuleInitializer moduleInitializer = new TestableModuleInitializer();
            SiteMapBuilderService siteMapBuilder = new SiteMapBuilderService();

            moduleInitializer.RegisterSiteMapInformation(siteMapBuilder);

            SiteMapNodeInfo node = siteMapBuilder.GetChildren(siteMapBuilder.RootNode.Key)[0];
            Assert.AreEqual("$ModuleName$", node.Key);
        }

    }

    class TestableModuleInitializer : $ModuleName$ModuleInitializer
    {
        public new void RegisterSiteMapInformation(ISiteMapBuilderService siteMapBuilder)
        {
            base.RegisterSiteMapInformation(siteMapBuilder);
        }
    }
}

File: sources/WCSF2010/Trunk/Source/WebClientFactoryGP/WebClientFactoryPackage/Templates/Projects/Module.Tests.VB/ModuleInitializerFixture.vb
Imports System
Imports System.Text
Imports System.Collections.Generic
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Microsoft.Practices.CompositeWeb
Imports Microsoft.Practices.CompositeWeb.Services
Imports Microsoft.Practices.CompositeWeb.Interfaces

<TestClass()> _
Public Class $ModuleName$ModuleInitializerFixture

    Public Sub New()

    End Sub

    <TestMethod()> _
    Public Sub $ModuleName$RegisteredOnSiteMap()

        Dim ModuleInitializer As TestableModuleInitializer = New TestableModuleInitializer()
        Dim siteMapBuilder As SiteMapBuilderService = New SiteMapBuilderService()

        ModuleInitializer.RegisterSiteMapInformation(siteMapBuilder)

        Dim node As SiteMapNodeInfo = siteMapBuilder.GetChildren(siteMapBuilder.RootNode.Key)(0)

        Assert.AreEqual("$ModuleName$", node.Key)

    End Sub

End Class

Public Class TestableModuleInitializer
    Inherits $ModuleName$ModuleInitializer

    Public Shadows Sub RegisterSiteMapInformation(ByVal siteMapBuilder As ISiteMapBuilderService)
        MyBase.RegisterSiteMapInformation(siteMapBuilder)
    End Sub

End Class
